### PR TITLE
Improving trigger performance and ability to change triggers for older projects

### PR DESCRIPTION
**Pull request by pedrocamargo**

### CODE REVIEW

**Comment by pedrocamargo** [comment]
This is really for you, @arobrien

File: aequilibrae/project/project_creation.py [hunk 1]
```diff
@@ -64,10 +64,12 @@ def remove_triggers(conn: Connection) -> None:
             for qry in cmd.split("\n"):
                 if qry[:2] == '--':
                     continue
+                while '  ' in qry:
```
Comment (arobrien) [inline aequilibrae/project/project_creation.py:67]: This is more elegantly solved with regular expressions, so we can build in extra whitespace, and case insensitivity into the regex
```python
# you can cmpile the regex outside the for loop outside the for loop
# we can limit this to the start of a line only if appropriate by using carat '^' in the regex like r'^\s+create....
create_drop_regex = re.compile(r'create\s+trigger\s+(\w+)', flags=re.I)

# search and capture trigger name in one step
m = re.search(create_drop_regex, qry)
if m:
    try:
        curr.execute(f'drop trigger if exists {m.group(1).lower()}')
    ....
```

File: aequilibrae/project/database_specification/triggers/network_triggers.sql [hunk 2]
```diff
@@ -190,13 +190,13 @@ create trigger updated_link_geometry after update of geometry on links
...
-    delete from nodes
-    where node_id not in (
-      select a_node
-      from links
-      union all
-      select b_node
-      from links);
+	Delete from Nodes
+    where node_id = old.a_node and
+           (select count(*) from Links where a_node = old.a_node or b_node = old.a_node) < 1;
```
Comment (arobrien) [inline aequilibrae/project/database_specification/triggers/network_triggers.sql:195]: good optimisation and cleanup here.

**Review by arobrien — COMMENTED** [review]
Looks good, but a couple of suggestions to clean up how the drops are processed,

File: aequilibrae/project/project_creation.py (hunk 1 above)
Comment (arobrien) [reply aequilibrae/project/project_creation.py:67]: I re-read the documentation, it looks like the re mondule caches recent regexes, so you don't need to bother pre-compiling them.

Comment (pedrocamargo) [reply aequilibrae/project/project_creation.py:68]: Done and tested

**Pull request merged**